main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1573,6 +1577,19 @@ object Lowering {
     LoweredAst.Expression.Tuple(exps, tpe, pur, eff, loc)
   }
 
+  /**
+    * Returns an InvokeStaticMethod expression to compile the regex pattern `exp`.
+    */
+  private def mkRegex(patt: String, loc: SourceLocation): LoweredAst.Expression = {
+    val cls = Class.forName("java.util.regex.Pattern")
+    val mtd = cls.getDeclaredMethod("compile", Class.forName("java.lang.String"))
+    val exp = LoweredAst.Expression.Cst(Ast.Constant.Str(patt), Type.Str, loc)
+    val tpe = Type.mkNative(cls, loc)
+    val pur = Type.Pure
+    val eff = Type.Empty
+    LoweredAst.Expression.InvokeStaticMethod(mtd, exp :: Nil, tpe, pur, eff, loc)
+  }
+
```

Comment:
Ah-- some code is also missing: You need to replace the Regex type with a Native type everywhere it may occur... There should be a `visitType` somewhere.

Reply:
Thanks Magnus - that was probably my next sticking point as I couldn't define a `ToString` instance.